### ISSUE TITLE
Error in syntax highlighting

### ISSUE THREAD

**Issue by birdonfire** [atom/language-python#110]
There's a problem with the syntax highlighting when viewing this file: https://github.com/18F/rdbms-subsetter/blob/master/subsetter.py

Something in the docstring at the top of the document is causing the problem. Perhaps something to do with the RST markup?

The syntax highlighting on the GitHub page suffers the same problem.

There are no actual syntax errors in the document, and the application code runs just fine.

SublimeText's highlighter renders it fine.

**Comment by birdonfire** [comment]
If I lower-case the word "INSERTS" on line 43, it totally fixes the highlighting.

**Comment by birdonfire** [comment]
It seems that the highlighter interprets the upper case INSERTS as the beginning of a SQL block that isn't then closed.

```
<div class="line cursor-line" data-screen-row="42">
    <span class="source python">
        <span class="string quoted double block sql python">
            rdbms-subsetter only performs the
            INSERTS; it
            <span class="string quoted single sql">
                <span class="punctuation definition string begin sql">'</span>
                s your responsibility to set
            </span>
        </span>
    </span>
    <span class="invisible-character">¶</span>
</div>
```

If I lower-case INSERTS, this block now looks like:
```
<div class="line cursor-line" data-screen-row="42">
    <span class="source python">
        <span class="string quoted double block sql python">
            rdbms-subsetter only performs the inserts; it's your responsibility to set
        </span>
    </span>
    <span class="invisible-character">¶</span>
</div>
```

**Comment by amlinger** [comment]
I'm experiencing the same issue, where we are extracting docstrings for usage in our API documentation, and therefore we are using markdown formatting inside the docstrings:
```python
"""Delete Resource
`DELETE http://some-url.com/resource`
"""
def destroy(self):
    pass
```
Which messes up the rest of the file (and here as well, where I expected `pass` to be highlighted, and `DELETE` not to be). 

To me it seems as the rule named [`string.quoted.double.block.sql.python`](https://github.com/atom/language-python/blob/master/grammars/python.cson#L1139-L1169) is the cause of this behaviour, and I've been fiddling around with it for a while but haven't really understood the grammar definitions yet. 

It seems as the rule accepts additional characters apart from only whitespace between the intial `"""` specified by begin, and the beginning of the SQL statement specified by the keywords + whitespace, but have not come any farther than that.

**Comment by hvdklauw** [comment]
Any progress on this?

I just ran into the same problem.

Seems the [syntax file](https://github.com/atom/language-python/blob/master/grammars/python.cson#L1154) starts the SQL context and only closes it when it encounters three double quotes, which basically eats them and leaves nothing for the commend to be closed.

**Comment by morenoh149** [comment]
I'm getting this for `DELETE` as well
```python
):
    '''
    Accounts
    ###**Actions**:
      - `GET /:id/card`
      - `POST /:id/card`
      - `PUT /:id/`
      - `DELETE /:id/`
    '''
    serializers = {
```

**Comment by krzysztof-sikorski** [comment]
I just stumbled upon this, it seems that
```python
"""'SELECT'"""
```
is enough to trigger that bug.

**Comment by dborstelmann** [comment]
I figured out the cause, and am working on a solution but would love ideas.  It's because when the rule is applied for the triple quote sql strings, it runs `include: source.sql` inside the matched pattern.  That is why the html classes that show on the text after a single `'` or `"` come from the language-sql grammar.  So the actual root cause is the language-sql grammar seeing the beginning of a sql string and then never finishing itself.  If anybody knows how to ignore the internal SQL strings inside of certain parts of triple quoted strings, let me know.

**Comment by jdw1996** [comment]
@dborstelmann Your proposed solution sounds reasonable to me, and I would definitely find it helpful.

**Comment by nckswt** [comment]
@dborstelmann have you managed to write up your solution yet? It would definitely be useful!

**Comment by dborstelmann** [comment]
@nckswt @jdw1996 Sorry I've been super busy recently, it looks the above reference will fix it?  Does that work for you guys?

**Comment by 50Wliu** [comment]
@dborstelmann True, the above PR should fix it, but some things to be aware of:
1. It's probably going to take a while to be reviewed,
2. I'm not even sure if that will get merged,
3. We're probably still going to switch over to MagicPython in at least the short term

**Comment by jdw1996** [comment]
@dborstelmann I didn't look too closely into that PR, but on @50Wliu's suggestion, I've now switched to MagicPython.

**Comment by nckswt** [comment]
Swapping over to MagicPython to try it out as well. Thanks!

**Comment by dborstelmann** [comment]
Lemme know if you guys still want this, but for now I will assume MagicPython is a good stopgap.

**Comment by all3fox** [comment]
Thank you, the highlight issue happened for me too because of a capitalized `WITH` in a docstring. Switching to [MagicPython][1] did the trick for me (just install and it works).

[1]: https://atom.io/packages/magicpython

**Comment by Hellzed** [comment]
MagicPython helped me too ! (In my case a SOFT_DELETE in a docstring)

**Comment by ezarowny** [comment]
The SQL keywords can actually appear anywhere within a word within a docstring. Example: `` `ALL_ITEMS_WITH_DATA` `` actually breaks highlighting for me. Of interest, it seems to break things in a different manner than just `ALL_ITEMS_WITH_DATA` does.

**Comment by goi42** [comment]
Just to pile on, I experience the same issue:
![screenshot 2017-12-22 10 21 30](https://user-images.githubusercontent.com/15774838/34292617-e9678caa-e701-11e7-9ca1-2893459ece8a.png)
MagicPython fixed it, but this really shouldn't be an issue...

**Comment by 7inyu** [comment]
Experiencing the same issue here. Someday...someday...
![screen shot 2018-04-17 at 7 47 21 pm](https://user-images.githubusercontent.com/6680385/38909130-2b7d2a6a-4278-11e8-98e0-ca4e5ba088b2.png)

**Comment by emptyopen** [comment]
Bumping...

![image](https://user-images.githubusercontent.com/18575100/41055893-4fe3184e-6977-11e8-9baf-6d18d68a0013.png)
